### PR TITLE
CIWEMB-289: Add Create credit note form

### PR DESCRIPTION
**Pull request by olayiwola-compucorp**
## Overview
This pull request adds a page to create credit note.

## Before
The page to create credit note did not exist. 

## After
The page to create new credit notes now exists.
errr

## Technical Details
The added Angular module `fe-creditnote` enables the creation of credit notes, and this module will also be used in the future for other credit note features that are more suitable for Angular templates than Smarty. The `CreditNoteAngular` class in `CRM/Financeextras/Page/Contribute/CreditNoteAngular.php` sets up the Angular environment and handles common page functionalities

The `templates/CRM/Financeextras/Page/Contribute/CreditNoteAngular.tpl` is responsible for rendering the Angular template for the credit note create form.

### CODE REVIEW

File: CRM/Financeextras/Page/Contribute/CreditNoteAngular.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * Class CRM_Financeextras_Page_Contribute_CrediitnotesAngular.
+ *
+ * Define an Angular base-page for Crediitnotes Module.
+ */
+class CRM_Financeextras_Page_Contribute_CreditNoteAngular extends \CRM_Core_Page {
+
+  /**
+   * Run Function.
+   *
+   * This function takes care of all the things common to all
```
Comment (omarabuhussein) [inline CRM/Financeextras/Page/Contribute/CreditNoteAngular.php:13]: You can just replace the docblock here with `@inheritdoc`

File: ang/fe-creditnote.ang.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<?php
```
Comment (omarabuhussein) [inline ang/fe-creditnote.ang.php:1]: is it possible to move these 3 files (fe-creditnote.ang.php, fe-creditnote.css and fe-creditnote.js) also under `fe-creditnote` ? or will things not work like that ?

File: CRM/Financeextras/Page/Contribute/CreditNoteAngular.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+/**
+ * Class CRM_Financeextras_Page_Contribute_CrediitnotesAngular.
```
Comment (omarabuhussein) [inline CRM/Financeextras/Page/Contribute/CreditNoteAngular.php:4]: CrediitnotesAngular -> CreditNoteAngular

File: CRM/Financeextras/Page/Contribute/CreditNoteAngular.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * Define an Angular base-page for Crediitnotes Module.
```
Comment (omarabuhussein) [inline CRM/Financeextras/Page/Contribute/CreditNoteAngular.php:6]: CrediitnotesAngular -> CreditNoteAngular

**Review by omarabuhussein — CHANGES_REQUESTED** [review]
left some notes, also:

 I don't think the PR title, the commit message or the "before" and "after" section should mention anything about Angular, given it is an implantation detail, these sections and parts should be more toward what the users sees and experience (in case of non technical tasks), so can you update them with that in mind ? thanks

File: ang/fe-creditnote.ang.php (hunk 2 above)
Comment (olayiwola-compucorp) [reply ang/fe-creditnote.ang.php:1]: No, it's not possible. but I've moved the fe-creditnote.css  file into a separate folder

Comment (olayiwola-compucorp) [reply ang/fe-creditnote.ang.php:1]: Behold the doc https://docs.civicrm.org/dev/en/latest/framework/angular/files/

**Review by omarabuhussein — APPROVED** [review]

**Pull request merged**